frontend/views/containers/settings/UserProfile.vue

```diff
@@ -170,43 +152,47 @@ export default {
       sbp('okTurtles.events/emit', OPEN_MODAL, mode)
       return false
     },
-    async save () {
-      if (this.ephemeral.newPicture) {
-        try {
-          this.form.picture = await imageUpload(this.form.picture)
-          this.ephemeral.newPicture = false
-        } catch (error) {
-          console.error(error)
-          this.ephemeral.errorMsg = L('Failed to upload user picture')
-          return false
+    async fileChange (fileList) {
+      if (!fileList.length) return
+      const fileReceived = fileList[0]
+      let picture
+
+      this.$refs.picture.setFromBlob(fileReceived)
+
+      try {
+        picture = await imageUpload(fileReceived)
+      } catch (error) {
+        console.error(error)
+        this.$refs.pictureMsg.danger(`${L('Failed to upload avatar. Please, try again.')} ${error.message}`)
+        return false
+      }
+
+      await this.setAttributes({ picture }, 'pictureMsg', L('Picture updated!'))
+    },
+    async saveProfile () {
+      this.$refs.formMsg.clean()
+      const attrs = {}
+
+      for (const key in this.form) {
+        if (this.form[key] !== this.attributes[key]) {
+          attrs[key] = this.form[key]
         }
       }
 
+      await this.setAttributes(attrs, 'formMsg', L('Your changes were saved!'))
+    },
+    async setAttributes (attrs, refFeedback, successMsg) {
       try {
-        this.ephemeral.profileSaved = false
-        const attrs = {}
-        for (const key in this.form) {
-          if (this.form[key] !== this.attributes[key]) {
-            attrs[key] = this.form[key]
-          }
-        }
         const attributes = await sbp('gi.contracts/identity/setAttributes/create',
           attrs,
           this.$store.state.loggedIn.identityContractID
         )
         await sbp('backend/publishLogEntry', attributes)
-        this.ephemeral.profileSaved = true
-      } catch (ex) {
-        // TODO: handle better
-        console.error(ex)
-        this.ephemeral.errorMsg = L('Failed to Save Profile')
+        this.$refs[refFeedback].success(successMsg)
+      } catch (error) {
+        console.error(error)
+        this.$refs[refFeedback].danger(`${L('Something went wrong, please try again.')} ${error.message}`)
       }
-    },
-    fileChange (fileList) {
-      if (!fileList.length) return
-      this.form.picture = fileList[0]
-      this.ephemeral.newPicture = true
```

Comment:
This `newPicture` variable was removed, and it was useful in preventing the same file from being uploaded twice accidentally by a user clicking the upload button twice.

Reply:
> being uploaded twice accidentally by a user clicking the upload button twice. 

How can the user do that twice? 🤔 I think it doesn't make sense anymore.